src/solvers/dg/dg_2d_parallel.jl

```diff
@@ -18,6 +18,25 @@ mutable struct MPICache
 end
 
 
+function MPICache()
+  mpi_neighbor_ranks = Vector{Int}(undef, 0)
+  mpi_neighbor_interfaces = Vector{Vector{Int}}(undef, 0)
+  mpi_send_buffers = Vector{Vector{Float64}}(undef, 0)
+  mpi_recv_buffers = Vector{Vector{Float64}}(undef, 0)
+  mpi_send_requests = Vector{MPI.Request}(undef, 0)
+  mpi_recv_requests = Vector{MPI.Request}(undef, 0)
+  n_elements_by_rank = OffsetArray(Vector{Int}(undef, 0), 0:-1)
+  n_elements_global = 0
+  first_element_global_id = 0
+
+  MPICache(mpi_neighbor_ranks, mpi_neighbor_interfaces,
+           mpi_send_buffers, mpi_recv_buffers,
+           mpi_send_requests, mpi_recv_requests,
+           n_elements_by_rank, n_elements_global,
+           first_element_global_id)
+end
+
+
```

Comment:
I tried an approach similar to `init_interfaces`/`init_interfaces!` and the other init functions here. The init function creates an empty `MPICache` and fills it with the `init_mpi_cache!` function that can then be reused for re-initializing in `amr_dg2d.jl`.
I am not sure if that is the best approach and if that's the best way to implement this approach.

Reply:
I am not sure if it makes sense right now. AFAICT, you never actually reuse the data structures in `init_mpi_cache!`. That is, all data structures are re-created anyways.

Reply:
As we discussed in person, we will leave it like this for now and see if we can reuse the existing structures in the future.